earth_enterprise/src/fusion/dbmanifest/dbmanifest.cpp

```diff
@@ -327,6 +332,64 @@ void DbManifest::GetManifest(geFilePool &file_pool,
   GetXmlManifest(stream_manifest);
 }
 
+// Read POI file and extract the data file names from within it and then add
+// them to the manifest too
+void DbManifest::GetPoiDataFiles(const std::string& poi_file,
+                     std::vector<ManifestEntry>* stream_manifest,
+                     std::vector<ManifestEntry>* search_manifest)
+{
+  notify(NFY_DEBUG,
+        "Parsing POI file %s looking for data files", poi_file.c_str());
+  size_t stream_poi_file_idx = stream_manifest->size() - 1;
```

Comment:
Is the POI file guaranteed to be the last entry in the list of manifests?  Also, are we certain there is only one?

Reply:
All the places that call this method currently call it right after adding the POI file to the manifest so right now yes both assumptions are true.  But I could see this being a less then obvious requirement and someone making a change that expectantly broke this requirement.  Any ideas to make it more obvious?  I could comment the assumptions to bring more attention them.  I could also move more of the code into this method and do the actual adding of the POI file in this method too.  Thoughts?

Reply:
Here are a couple of thoughts I had.  I'll leave it up to you to decide how you want to handle it.

1. Instead of passing the full `stream_manifest` and `search_manifest` vectors, just pass the new manifest entry.  So call it like `GetPoiDataFiles(curr, stream_manifest.back(), search_manifest.back())`. That makes the requirement more obvious - you add the element to the back and then get the back element in the next line, which is clear enough for me.

2. Have `GetPoiDataFiles` return the list of dependencies and call it first.  So the code ends up something like
```
         std::vector<std::string> deps = GetPoiDataFiles(curr);
         if (stream_manifest != search_manifest) {
           stream_manifest->push_back(ManifestEntry(orig, curr, deps));
         }
         search_manifest->push_back(ManifestEntry(orig, curr, deps));
```
Then you have to worry about the cost of copying a vector, though.

Reply:
Now that I think about it your option 1 is better than what  I was proposing.  The only reason I'm passing the manifest vectors is because originally we where adding the data files to the manifest vectors.  But we had to change that and given that change passing the manifest entries is a much cleaner solution.